packages/docs/community/release-process/smart-contracts.md

```diff
@@ -5,93 +5,110 @@ This release process is a work in progress. Many infrastructure components requi
 {% endhint %}
 
 ## Versioning
+
 Each deployed Celo core smart contract is versioned independently, according to semantic versioning, as described at [semver.org](https://semver.org), with the following modifications:
 
-  * STORAGE version when you make incompatible storage layout changes
-  * MAJOR version when you make incompatible ABI changes
-  * MINOR version when you add functionality in a backwards compatible manner, and
-  * PATCH version when you make backwards compatible bug fixes.
+- STORAGE version when you make incompatible storage layout changes
+- MAJOR version when you make incompatible ABI changes
+- MINOR version when you add functionality in a backwards compatible manner, and
+- PATCH version when you make backwards compatible bug fixes.
+
+Changes to core smart contracts are made via on-chain Governance, approximately four times a year. When a release is made, **all** smart contracts from the release branch that differ from the deployed smart contracts are released, and included in the **same** governance proposal. Each release is identified by a unique monotonically increasing version number `N`, with `1` being the first release.
+
+### Core Contracts
 
-Changes to core smart contracts are made via on-chain Governance, approximately four times a year. When a release is made, **all** smart contracts from the release branch that differ from the deployed smart contracts are released, and included in the **same** governance proposal.
+Every deployed Celo core contract has its current version number as a constant which is publicly accessible via the `getVersion()` function, which returns the storage, major, minor, and patch version. Version number is encoded in the Solidity source and updated as part of code changes.
```

Comment:
I moved this paragraph back up since it seemed closer related to the versioning paragraph above